### PR TITLE
Fix compatibility problems with a pre-Cohen save

### LINKED ISSUE #3039: Green & Grassmann OK but Gröbner crashes with old RetroRendezvous save
Green & Grassmann successfully load in KSP 1.11.2 the [old save](https://www.dropbox.com/s/d9vbdw0wba0jkxv/rendezvous.zip) at:

https://forum.kerbalspaceprogram.com/index.php?/topic/161390-principia-challenge-munar-retrograde-rendezvous-122/

However I get the following crash when trying to open the same save in KSP 1.11.2 with Gröbner: 

[Principia_ERROR_20210625_223123_15380.zip](https://github.com/mockingbirdnest/Principia/files/6718957/Principia_ERROR_20210625_223123_15380.zip)

`E0625 22:31:23.531883 22048 interface.cpp:691] Principia version 2021061011-Gröbner-0-g05ed8f86927035b5820d611536d711b3176db3d4 built on 2021-06-05T16:53:53Z by Microsoft Visual C++ version 192930037 for Windows x86-64

F0625 22:31:40.448882 22072 continuous_trajectory_body.hpp:489] Check failed: (continuous_trajectory->checkpointer_->ReadFromCheckpointAtOrBefore( desired_t_min)) == ::absl::OkStatus() (NOT_FOUND: No checkpoint vs. OK)`

### PR DESCRIPTION
**Pull request by pleroy**
While I am at it I am reviving the compatibility test to make sure that this doesn't regress.  The test save is relatively small at 1 828 824 bytes.

Fix #3039.

### CODE REVIEW

File: ksp_plugin_test/interface_compatibility_test.cpp [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+using base::PushDeserializer;
+using ksp_plugin::Plugin;
+using testing_utilities::ReadLinesFromBase64File;
+using testing_utilities::ReadLinesFromHexadecimalFile;
+using ::testing::NotNull;
+
+const char preferred_compressor[] = "gipfeli";
+const char preferred_encoder[] = "base64";
+
+class InterfaceCompatibilityTest : public testing::Test {
```
Comment (eggrobin) [inline ksp_plugin_test/interface_compatibility_test.cpp:30]: Let us keep that in PluginCompatibilityTest; while this test is interface-only, we may want to inspect saves in other tests, e.g., https://github.com/eggrobin/Principia/blob/saves/ksp_plugin_test/plugin_compatibility_test.cpp.

File: ksp_plugin_test/interface_compatibility_test.cpp [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      principia__DeserializePlugin("",
+                                   &deserializer,
+                                   &plugin,
+                                   compressor.data(),
+                                   encoder.data());
+      LOG(ERROR) << "Deserialization complete";
+    }
+    EXPECT_THAT(plugin, NotNull());
+
+    // Write that plugin back to another file with the preferred format.
```
Comment (eggrobin) [inline ksp_plugin_test/interface_compatibility_test.cpp:63]: Split the reading and writing into separate files.

File: ksp_plugin_test/interface_compatibility_test.cpp (hunk 1 above)
Comment (pleroy) [reply ksp_plugin_test/interface_compatibility_test.cpp]: Done.

File: ksp_plugin_test/interface_compatibility_test.cpp (hunk 2 above)
Comment (pleroy) [reply ksp_plugin_test/interface_compatibility_test.cpp]: Done.

File: ksp_plugin_test/plugin_compatibility_test.cpp [hunk 3]
```diff
@@ -1,111 +1,134 @@
...
+    Plugin const* plugin = nullptr;
+
+    PushDeserializer* deserializer = nullptr;
+    auto const lines =
+        encoder == "hexadecimal" ? ReadLinesFromHexadecimalFile(filename)
+        : encoder == "base64"    ? ReadLinesFromBase64File(filename)
+                                  : std::vector<std::string>{};
+    CHECK(!lines.empty());
+
+    LOG(ERROR) << "Deserialization starting";
```
Comment (eggrobin) [inline ksp_plugin_test/plugin_compatibility_test.cpp:50]: LOG(INFO) and google::LogToStdErr() in the constructor.

**Review by eggrobin — APPROVED** [review]

File: ksp_plugin_test/plugin_compatibility_test.cpp (hunk 3 above)
Comment (pleroy) [reply ksp_plugin_test/plugin_compatibility_test.cpp:50]: Nope.  When I do that the output gets swamped with messages emitted from various places within the plugin (and its subobjects).  I want to be able to the progress of the test, not random crud unrelated to the test.

**Pull request merged**